runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -66,7 +80,9 @@ TSHighlighter.hl_map = {
     ["type.builtin"] = "Type",
     ["structure"] = "Structure",
     ["include"] = "Include",
-}
+}, subcapture_fallback)
+
+TSHighlighter.hl_map.__index = subcapture_fallback
```

Comment:
I think I honestly liked your previous approach better without the intermediate local. Is there a reason why the current approach is better?

Reply:
I'd say this PR is ready. I can revert this

Reply:
you mean?
```
-- If @definition.special does not exist use @definition instead
function TSHighlighter.hl_map:__index(capture)
  local rtn
  local shortened = capture
  while not rtn and shortened do
    shortened = shortened:match('(.*)%.')
    rtn = shortened and rawget(self, shortened)
  end
  rawset(self, capture, rtn or "__notfound")
  return rtn
end
```

Reply:
I think you can just remove the last line where you set `__index` directly, right? And just do
```lua
local subcapture_fallback = {
 __index = function(capture)
   ...
 end
}
TSHighlighter.hl_map = setmetatable({
  ...
  }, subcapture_fallback)
```

Reply:
> you mean?
> ...

I don't think that works. I'd just remove the current line 85 (and possibly simplify the creation of the metatable itself as in my last comment).